src/UseCases/CancelMembershipApplication/CancelMembershipApplicationUseCase.php

```diff
@@ -27,7 +31,63 @@ public function __construct( MembershipApplicationAuthorizer $authorizer,
 	}
 
 	public function cancelApplication( CancellationRequest $request ): CancellationResponse {
-		// TODO
+		$application = $this->getApplicationById( $request->getApplicationId() );
+
+		if ( $application === null ) {
+			return $this->newFailureResponse( $request );
+		}
+
+		$application->cancel();
+
+		try {
+			$this->repository->storeApplication( $application );
+		}
+		catch ( StoreMembershipApplicationException $ex ) {
+			// TODO: log?
+			return $this->newFailureResponse( $request );
+		}
+
+		$this->sendConfirmationEmail( $application );
```

Comment:
https://phabricator.wikimedia.org/T133161
The[ feature is in the old application](https://github.com/wmde/fundraising/blob/master/spenden/contact.inc.php#L334-L342), albeit with a bad wording of the message.